index.js

```diff
@@ -17,7 +25,44 @@ function isEmpty(obj){
     return true
 }
 
-function createXHR(options, callback) {
+function normalizeArgs(arg1, arg2, arg3) {
+    var options, callback;
+
+    if (typeof arg1 === "string") {
+        if (typeof arg2 === "object" && typeof arg3 === "function") {
+            // xhr(url, options, cb)
+            options = xtend(arg2)
+            options.uri = arg1
+            callback = arg3
+        } else {
+            // xhr(url, cb)
+            options = { uri: arg1 }
+            callback = arg2
+        }
+    } else {
+      // xhr(options, cb)
```

Comment:
The nice things about the way request does it is they just end up merging everything into a single object for their internal api to use.